src/internal/scheduled/scheduleIterable.ts

```diff
@@ -22,36 +23,29 @@ export function scheduleIterable<T>(input: Iterable<T>, scheduler: SchedulerLike
         iterator = (input as any)[Symbol_iterator]();
 
         // Schedule the first iteration and emission.
-        subscriber.add(
-          scheduler.schedule(function () {
-            // Check to make sure teardown was not triggered
-            // by the consumer since the last iteration.
-            if (!subscriber.closed) {
-              let value: T;
-              let done: boolean | undefined;
-              try {
-                // Pull the value out of the iterator
-                ({ value, done } = iterator.next());
-              } catch (err) {
-                subscriber.error(err);
-                return;
-              }
-              if (done) {
-                // If it is "done" we just complete. This mimics the
-                // behavior of JavaScript's `for..of` consumption of
-                // iterables, which will not emit the value from an iterator
-                // result of `{ done: true: value: 'here' }`.
-                subscriber.complete();
-              } else {
-                // The iterable is not done, emit the value.
-                subscriber.next(value);
-                // Reschedule. This will cause this function to be
-                // called again on the same scheduled delay.
-                this.schedule();
-              }
+        caughtSchedule(subscriber, scheduler, function () {
+          // Check to make sure teardown was not triggered
+          // by the consumer since the last iteration.
+          if (!subscriber.closed) {
```

Comment:
nitpick: This isn't part of the refactor - it's in the original code - but I'm not sure this `subscriber.closed` check is needed here. If the subscriber has been explicitly unsubscribed, so too will have been the scheduled action that was added to it. So the work function should never be called in this situation.

FWIW, the refactor in `src/internal/observable/generate.ts` removes the outer and inner checks of `subscriber.closed` - which I think is the correct thing to do, as the rescheduling `schedule` method checks `closed` internally - so I'm not sure why the outer check is needed here.